docs/developing.rst

```diff
@@ -3,14 +3,225 @@
+          The default implementation of this function should suffice for most developers.
+        - ``postprocess_system``: operates identically to ``create_force``, but is run after each ParameterHandlers' ``create_force`` has already been called.
+          The default implementation of this method simply does nothing, and should suffice for most developers.
+- ParameterType
+- ToolkitRegistry
+- Molecule.to/from_object
+- Force field directories
+
+Molecule definition
+'''''''''''''''''''
+
+Required stereochemistry
+''''''''''''''''''''''''
+
+Conformation dependence
+'''''''''''''''''''''''
+
+
+
+Reliance on external dependencies
+'''''''''''''''''''''''''''''''''
+
+
+
+ForceField file paths
+'''''''''''''''''''''
+
+Documentation
+'''''''''''''
+If you define a new class, add new files to autodoc
+
+User Experience
+'''''''''''''''
+
+One important aspect of how we make design decisions is by asking "who do we envision using this software, and what would they want it to do here?".
+There is a wide range of possible users, from non-chemists, to students/trainees, to expert computational medicinal chemists.
+We have decided to build functionality intended for use by `expert medicinal chemists`, and whenever possible, add fatal errors if the toolkit risks doing the wrong thing.
+So, for example, if a molecule is loaded with an odd ionization state, we assume that the user has input it this way intentionally.
+This design philosophy invariably has tradeoffs -- For example, the OFF Toolkit will give the user a hard time if they try to load a "dirty" molecule dataset, where some molecules have errors or are not described in enough detail for the toolkit to unambiguously parametrize them.
+If there is risk of misinterpreting the molecule (for example, bond orders being undefined or chiral centers without defined stereochemistry), the toolkit should raise an error that the user can override.
+In this regard we differ from RDKit, which is more permissive in the level of detail it requires when creating molecules.
+This makes sense for RDKit's use cases, as several of its analyses can operate with a lower level of detail about the molecules.
+Often, the same design decision is the best for all types of users, and there is no need for discussion.
+But when we do need to make tradeoffs, "assume the user is an expert" is our guiding principle.
+
+At the same time, we aim for "automagic" behavior whenever a decision will clearly go one way over another.
+System parametrization is an inherently complex topic, and the OFF toolkit would be nearly unusable if we required the user to explicitly approve every aspect of the process.
+For example, if a ``Topology`` has its ``box_vectors`` attribute defined, we assume that the resulting ``System`` should be periodic.
+
+
+
+Setting up a development environment
+""""""""""""""""""""""""""""""""""""
```

Comment:
Converting some notes from yesterday into partial instructions:

* Install conda
  * See https://www.anaconda.com/distribution/

* Set up conda environment
 
```
conda create --name off-dev -c conda-forge  -c omnia -c openeye openforcefield openeye-toolkits
conda remove --force openforcefield
git clone https://github.com/openforcefield/openforcefield
pip install -e .
```
* Obtain and store Open Eye license somewhere like `~/.oe_license.txt
  * Optionally store the path in environmental variable `OE_LICENSE`: `echo "export OE_LICENSE=/Users/yournamehere/.oe_license.txt" >> ~/.bashrc`

Reply:
An alternative approach that doesn't require installation and removal of `openforcefield` itself, if desired:
```
conda create --name off-dev -c conda-forge -c omnia -c openeye openeye-toolkits
conda install --name off-dev -c conda-forge -c omnia -c openeye --only-deps openforcefield

git clone https://github.com/openforcefield/openforcefield
cd openforcefield
pip install -e .
```

Reply:
I think either one is fine. The separate `create` and `install` in @dotsdl's suggestion is more graceful, but I fear it could lead to dependency-resolving complications if our recipe gets more complex and we have to install more packages.

Reply:
Need to add:
```
conda activate off-dev
```
After conda installs and before use of `pip`.